app/controllers/users/viewing_parties_controller.rb

```diff
@@ -1,5 +1,67 @@
 class Users::ViewingPartiesController < ApplicationController
+  before_action :load_host
+  before_action :load_movie
+  before_action :validate_duration, only: [:create]
```

Comment:
This was a clever way to use a callback to create a custom validation. We can talk tomorrow about how we could have implemented this in the model with our own custom validator object.

Reply:
I did try it in the model but since we do not have a movies table, it won't seem to work. We're getting the movie via a method and not a database relationship, so we can't validate on it.